backend/functions/src/utils/on-create-document.ts

```diff
@@ -0,0 +1,4 @@
+export function onCreateDocument(obj: object) {
+  const date: Date = new Date();
+  return Object.assign(obj, { updatedAt: date, createdAt: date, isDeleted: false });
```

Comment:
You should use modern operators and syntax when possible. You could use the `...` spread operator to duplicate the `obj` argument and add the new fields inside the object literal.